[datadog_workflow_automation] Add Workflows to Terraform (#2833)
DataDog/terraform-provider-datadog: docs/resources/workflow_automation.md

  tags        = ["service:foo", "source:alert", "team:bar"]
  published   = true

  spec_json = jsonencode(
    {
      "triggers" : [
        {
          "startStepNames" : [
            "Send_Email"
          ],
          "monitorTrigger" : {}
        }
      ],
      "steps" : [
        {
          "name" : "Send_Email",
          "actionId" : "com.datadoghq.email.send",
          "parameters" : [
            {
              "name" : "to",
              "value" : "REPLACE_ME"
            },
            {
              "name" : "subject",
              "value" : "Monitor \"{{ Source.monitor.name }}\" alerted"
            },
            {
              "name" : "message",
              "value" : "This message is from {{ WorkflowName }}. \n\nYou can find a link to the monitor here: {{ Source.url }}."
            }
          ],
          "display" : {
            "bounds" : {
              "x" : 0,
              "y" : 216
            }
          }
        }
      ],
      "handle" : "my-handle"
    }
  )
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Required

- `description` (String) Description of the workflow.
- `name` (String) Name of the workflow.

> - `published` (Boolean) Set the workflow to published or unpublished. Workflows in an unpublished state will only be executable via manual runs. Automatic triggers such as Schedule will not execute the workflow until it is published.

Reviewer: ```suggestion
- `published` (Boolean) Set the workflow to published or unpublished. Workflows in an unpublished state are only be executable through manual runs. Automatic triggers such as Schedule do not execute the workflow until it is published.
```

Author: Used `Set the workflow to published or unpublished. Workflows in an unpublished state are only executable through manual runs. Automatic triggers such as Schedule do not execute the workflow until it is published.`
